File: frontend/src/api/lut.ts
import client from "./client";

export interface LutModelInfo {
  name: string;
  path: string;
  format: string;
}

export interface LutRunRequest {
  dataset_id: string;
  image_ids?: string[];
  lut_path: string;
  intensity: number;
  replace: boolean;
  subfolder?: string;
}

export const lutApi = {
  models: () =>
    client.get<LutModelInfo[]>("/lut/models").then((r) => r.data),

  run: (req: LutRunRequest) =>
    client
      .post<{ job_id: string; total: number }>("/lut/run", req)
      .then((r) => r.data),
};

File: backend/schemas/lut.py
from pydantic import BaseModel, field_validator


class LutModelInfo(BaseModel):
    name: str
    path: str
    format: str


class LutRunRequest(BaseModel):
    dataset_id: str
    image_ids: list[str] | None = None
    lut_path: str
    intensity: float = 1.0
    replace: bool = False
    subfolder: str | None = None
    label: str | None = None

    @field_validator("intensity")
    @classmethod
    def _clamp_intensity(cls, v: float) -> float:
        return max(0.0, min(1.0, v))
